src/internal/util/subscribeTo.ts

```diff
@@ -0,0 +1,38 @@
+import { Observable, ObservableInput } from '../Observable';
+import { subscribeToArray } from './subscribeToArray';
+import { subscribeToPromise } from './subscribeToPromise';
+import { subscribeToIterable } from './subscribeToIterable';
+import { subscribeToObservable } from './subscribeToObservable';
+import { isArrayLike } from './isArrayLike';
+import { isPromise } from './isPromise';
+import { isObject } from './isObject';
+import { iterator as Symbol_iterator } from '../symbol/iterator';
+import { observable as Symbol_observable } from '../symbol/observable';
+import { Subscriber } from '../Subscriber';
+
+export const subscribeTo = <T>(result: ObservableInput<T>) => {
+  if (result instanceof Observable) {
+    return (subscriber: Subscriber<T>) => {
```

Comment:
how much perf will hurt if we use `subscribeToObservable` in here instead, and let `subscribeToObservable` have checker to scalar? (like
```
if (obs[symbol.observable) {
 if (!!obs._isScalar) {
 }
 else {
...
 }
}
```

Reply:
instanceof checks are really fast . Especially compared to property lookups. What would be the win if we did change this?

Reply:
not so much, it'll reduce some codes duplicated between symbolic observable to instance observable. Reason asking if it hurts perf - if it doesn't hurt much, maybe considerable.

Reply:
Oh... yeah, we don't want it to fall down to `subscribeToObservable`, that's meant to subscribe to "lowercase-o" observables where we can't assume anything about them. That means we have to do a call to `Symbol.observable` implementation on whatever the object is, and get out an unknown observable, which will only take an observer, and will only return a plain Subscription object, not what we expect inside of RxJS.

This `instanceof` check is basically saying "Hey, l know this implementation and I trust it, so do the fastest thing"... it's the 99% use case for this code.  I don't think I want to change it.

Reply:
makes sense.